### ISSUE TITLE
Todo

### ISSUE THREAD

**Issue by FGRibreau** [iadvize/mq-library#2]
```
// A la connexion :    
// amqp -> gérer la possibilité de créer plusieurs canaux 
// amqp -> créer echanges/queues si ça n'existe pas    
// amqp -> vérifier le type des bindings/queue & echange, si pas compatible => crash   

// En cas de mauvais nom de queue :    
// - proposer une correction orthographique (distance de levenshtein)s   
// - donner le nom réel de la queue    

// json-schema pour la documentation ?   

-## Gestion du diff de version entre la production le "master" sur dev    

-- Avant chaque test : loader le schéma RabbitMQ de production    
-- Lancer les tests (qui vont vérifier que le schema match)   
    + > crash du test    
    +   -> forcer le test en unstable    
    +   -> alerte le dev/devops qu'il doit une modification en production    
    +   -> une fois la modification faite    
    +       -> relancer la build
```

**Comment by remy-theroux** [comment]
Est-ce le rôle de la library de crasher le process et pas celui du service de savoir ce qu'il veut faire d'un problème de connexion @FGRibreau

**Comment by FGRibreau** [comment]
La lib va throw et donc le service peut très bien catch cette exceptions (ou avec les domain nodejs) pour choisir d'en faire autre chose

**Comment by remy-theroux** [comment]
Actuellement c'est la lib qui gère l'error, et elle kill le process
https://github.com/iadvize/mq-library/blob/master/lib/amqp.js#L151

**Comment by FGRibreau** [comment]
Ok ! 

Je n'ai pas review le code de @Freyskeyd, mais dans l'idée de toute façon on veut que par défaut ça crash le process, c'est la bonne approche et un good default.
Maintenant, on veut aussi laisser le choix aux utilisateurs de la lib d'override ce comportement, à voir dans une v0.1 donc :)

**Comment by Freyskeyd** [comment]
Pour la prochaine release, lorsqu'une error amqp sera levé, un event sera emit de type "connectionError", avec le type d'erreur. (Connection, assertion, handshake)

**Comment by JulienBreux** [comment]
Pas normal ça !

**Comment by FGRibreau** [comment]
@Freyskeyd plutôt que "connectionError" un event "error" serait plus standard en NodeJS.
Et ajouter un attr `code` à l'erreur avec `ERR_CONNECTION`, `ERR_ASSERTION`, `ERR_HANDSHAKE`

**Comment by Freyskeyd** [comment]
@FGRibreau yep nice!

Add it to the roadmap!

**Comment by jbpionnier** [comment]
Je viens de découvrir votre lib sympatique.

Elle fait le job, cependant ais-je raté une étape mais j'aurai voulu créer d'autre channel.
Comment faire pour obtenir la connection ? :-x

**Comment by JulienBreux** [comment]
@jbpionnier effectivement, pour le moment, c'est un manque dans la librairie, vous constaterez qu'à cette endroit là : https://github.com/iadvize/mq-library/blob/master/lib/amqp.js#L50-L52.
Nous considérons qu'il n'y à besoin que d'un seul et unique "channel".
Point que nous améliorerons ;)
